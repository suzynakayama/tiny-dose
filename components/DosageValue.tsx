import { StyleSheet, Text, Image, View } from 'react-native';
import React from 'react';
import SyringeImage from '@/assets/images/syringe.png';

// This component displays the calculated dosage value and maximum dose information.
// It contains the image of a syringe, the dosage value, and an optional maximum dose message.

interface DosageValueProps {
  dosage: string;
  maxDose: number | null;
}

export default function DosageValue({ dosage, maxDose }: DosageValueProps) {
  return (
    <View style={styles.dosageContainer}>
      <Image source={SyringeImage} style={styles.image} />
      <Text style={styles.dosageText}>Your child's dose:</Text>
      <Text style={styles.dosageValue}>{`${dosage}`}</Text>
      {maxDose && (
        <Text
          style={styles.maxDose}
        >{`⚠️ Do not exceed ${maxDose} doses in 24 hours.`}</Text>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  dosageContainer: {
    flex: 1,
    alignItems: 'center',
    marginBottom: 10,
  },
  image: {
    maxWidth: 180,
    height: 115,
  },
  dosageText: {
    fontSize: 16,
    marginBottom: 10,
  },
  dosageValue: {
    fontSize: 20,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  maxDose: {
    fontSize: 14,
    marginTop: 15,
    fontStyle: 'italic',
  },
});
