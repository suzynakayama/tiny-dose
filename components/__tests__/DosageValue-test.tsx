import React from 'react'
import { render, screen } from '@testing-library/react-native'
import DosageValue from '../DosageValue'

test('renders correctly with dosage and max dose', () => {
  const dosage = '5 ml'
  const maxDose = 4

  const dosageValueComponent = render(<DosageValue dosage={dosage} maxDose={maxDose} />)

  const dosageText = screen.getByText("Your child's dose:")
  expect(dosageText).toBeOnTheScreen()

  const dosageValue = screen.getByText(dosage)
  expect(dosageValue).toBeOnTheScreen()

  const maxDoseText = screen.getByText(`⚠️ Do not exceed ${maxDose} doses in 24 hours.`)
  expect(maxDoseText).toBeOnTheScreen()

  const dosageValueJson = dosageValueComponent.toJSON()
  expect(dosageValueJson).toMatchSnapshot()
})

test('renders correctly without max dose', () => {
  const dosage = '10 ml'
  const maxDose = null

  const dosageValueComponent = render(<DosageValue dosage={dosage} maxDose={maxDose} />)

  expect(screen.queryByText(/Do not exceed/)).toBeNull()

  // Snapshot testing
  const dosageValueJson = dosageValueComponent.toJSON()
  expect(dosageValueJson).toMatchSnapshot()
})
