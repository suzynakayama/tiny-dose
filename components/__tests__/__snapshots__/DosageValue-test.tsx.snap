// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with dosage and max dose 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 10,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/syringe.png",
      }
    }
    style={
      {
        "height": 115,
        "maxWidth": 180,
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    Your child's dose:
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    5 ml
  </Text>
  <Text
    style={
      {
        "fontSize": 14,
        "fontStyle": "italic",
        "marginTop": 15,
      }
    }
  >
    ⚠️ Do not exceed 4 doses in 24 hours.
  </Text>
</View>
`;

exports[`renders correctly without max dose 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 10,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/syringe.png",
      }
    }
    style={
      {
        "height": 115,
        "maxWidth": 180,
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    Your child's dose:
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    10 ml
  </Text>
</View>
`;
